lumispy/utils/axes.py

```diff
@@ -176,6 +179,179 @@ def var2invcm(variance, factor, invcmaxis, ax0=None):
     return variance * (factor / (invcmaxis**2)) ** 2
 
 
+#
+# navigation axis manipulation
+#
+
+
+def crop_edges(
+    S,
+    crop_range=None,
+    crop_units="pixels",
+    rebin_nav=False,
+    *,
+    crop_px=None,
+):
+    """
+    Cropping along the navigation axes of the a list of signal objects.
+    Crop the amount of pixels from the four edges of the scanning
+    region, from the edges inwards. Cropping can happen uniformly on all
+    sides or by specifying the cropping range for each axis or each side. If the navigation axes shape is different, all signals can be rebinned to match the shape of the first signal in the list.
+
+    Parameters
+    ----------
+    S : list of HyperSpy Signal objects with the same navigation axes or a single HyperSpy Signal object.
+    crop_range : int, float, tuple
+        Number of pixels or percentage (between 0 and 1) of image width/height to be cropped.
+        If a number or a tuple of size 1 is passed, all sides are cropped by the
+        same amount. If a tuple of size 2 is passed (``crop_x``, ``crop_y``), a different
+        amount of pixels/percentage is cropped from the x and y directions,
+        respectively. If a tuple of size 4 is passed (``crop_left``, ``crop_bottom``,
+        ``crop_right``, ``crop_top``), a different amount of pixels/percentage is
+        cropped from each edge individually.
+
+    crop_units : str
+        Select in which units cropping happens. Value can be either ``pixels``/``px`` (default),
+        or ``percent``/``%``. All units are rounded downwards.
```

Comment:
The API could be simplify by merging the `crop_ranges` and `crop_units` arguments, similarly as this is done with the [`inav` and `isig`](https://hyperspy.readthedocs.io/en/stable/user_guide/signal.html#indexing):
- use `int` for indices (pixel)
- use `float` to use the axis values.
- use string starting with `rel`, e.g. `rel0.1` to use relative coordinate, i.e. the equivalent of percent
- use string finishing with an unit to to unit conversion on the fly.

This is in implemented in the axis, so this should be easy to re-use it.

Reply:
See [crop_signal1D](https://hyperspy.readthedocs.io/en/latest/api/hyperspy._signals.signal1d.html#hyperspy._signals.signal1d.Signal1D.crop_signal1D) on how this is implemented in a similar function (just signal instead of navigation space) in HyperSpy.

An interactive option could be interesting as well.

Reply:
Thanks for the comments. 
The `crop_signal1D` and the respective 2d equivalent `crop_image` do not accept sting fancy indexing (e.g. `rel0.1` nor `1nm`). These functions only work with floats or integers.
I can implement them, but the fancy string formatting would not work (which I think it's ok).

Also, I'd like to point out that this function here does a few more checks than the basic signal operators (such as checking for axes shapes, guessing what the cropping ranges should be, and rebins...).